### ISSUE TITLE
Error on querying after hours of idle

### ISSUE THREAD

**Issue by lideming** [manyuanrong/deno_mysql#79]
```
Error: Read packet timeout
    at Connection.nextPacket (connection.ts:101:17)
    at async Connection.execute (connection.ts:162:19)
    at async Connection.query (connection.ts:143:20)
    at async client.ts:90:14
    at async Client.useConnection (client.ts:111:22)
    at async Client.query (client.ts:89:12)
    at (my code...)
```

To reproduce:
1) Create a client with connection pool
2) Do a query
3) Wait for more than 8 hours (the default mysql server idle timeout)
4) Query again, get the error

The case is that mysql server disconnect the connection in the pool on idle timeout.
Then we get the disconnected connection from the pool, read from it, and get the EOF.

I suggest to add client-side connection idle timeout, which should shorter than 8 hours by default.

**Comment by BayoKwendo** [comment]
I am also experiencing the same issue, lets wait for hope atleast

**Comment by lideming** [comment]
@BayoKwendo 
I have created a PR. Welcome to try my fix now.
If you are sure this is the same issue, you can close #81 ;)

**Comment by BayoKwendo** [comment]
@lideming what next

**Comment by lideming** [comment]
@BayoKwendo Wait for merging...
If you want to try it now, import from `https://raw.githubusercontent.com/lideming/deno_mysql/idle_timeout/mod.ts`

**Comment by BayoKwendo** [comment]
@lideming  actually its working perfect

**Comment by BayoKwendo** [comment]
@lideming  i have experienced another issue using the above import  "Connection closed unexpectedly"  after idle. Any clue why that happens

**Comment by lideming** [comment]
@BayoKwendo 
Before my PR, reading from disconnected connection will throw the read timeout error. This is incorrect and I have changed it to throw "Connection closed unexpectedly".

There are three settings:
`interactive_timeout` and `wait_timeout` (in seconds, usually both are the same value) in MySQL server configuration, and
`idleTimeout` in the Deno client, which is added in my PR.

Make sure the client setting are shorter than the server setting.

AFAIK, MySQL server default timeout is 8 hours, so I set the default `idleTimeout` to 4 hours. If the server timeout setting is changed, `idleTimeout` should also be changed accordingly.

If you confirm the error still happen while the settings are correct, then unfortunately there is another issue.
